DashKitUI/Views/DashSettingsView.swift

```diff
@@ -59,15 +129,81 @@ struct DashSettingsView<Model>: View where Model: DashSettingsViewModelProtocol
                 }.padding(.bottom, 8)
             }
             
-            Section(header: Text("Pod").font(.headline).foregroundColor(Color.primary)) {
+            if self.viewModel.havePod {
+                Section(header: Text("Pod").font(.headline).foregroundColor(Color.primary)) {
+                    self.viewModel.lifeState.deliveryState.map { (deliveryState) in
+                        HStack {
+                            Button(action: {
+                                self.viewModel.suspendResumeTapped()
+                            }) {
+                                Text(deliveryState.suspendResumeActionText)
+                                    .foregroundColor(deliveryState.suspendResumeActionColor)
+                            }
+                            Spacer()
+                            if deliveryState.transitioning {
+                                ActivityIndicator(isAnimating: .constant(true), style: .medium)
+                            }
+                        }
+                    }
+                }
+
+                Section() {
+                    
+                    NavigationLink(destination: PodDetailsView(podDetails: self.viewModel.podDetails)) {
+                        Text("Pod Details").foregroundColor(Color.primary)
+                    }
+                        
+                    self.viewModel.lifeState.activatedAt.map { (activatedAt) in
+                        HStack {
+                            Text("Pod Insertion")
+                            Spacer()
+                            Text(self.dateFormatter.string(from: activatedAt))
+                        }
+                    }
+
+                    self.viewModel.lifeState.activatedAt.map { (activatedAt) in
```

Comment:
Sorry, this confused me.  Why are there two clauses of 
```
self.viewModel.lifeState.activatedAt.map { (activatedAt) in
```
?  
Can they be combined, or might this be a copy/paste error?

(P.S. I still would love to replace this `.map { ... }` jazz with a handy `ifNotNull` function, but that's another thing...)

Reply:
The way `Section` works with this function builder requires that they evaluate to separate views.  As for the `map` notation, I think once function builders allow local lets this can be changed to something that looks more conventional to you.